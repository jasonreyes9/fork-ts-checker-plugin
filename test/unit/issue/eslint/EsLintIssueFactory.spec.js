const {
  createIssuesFromEsLintReports
} = require('../../../../lib/issue/eslint/EsLintIssueFactory');

describe('[UNIT] issue/eslint/EsLintIssueFactory', () => {
  const ES_LINT_MESSAGE_ERROR = {
    column: 0,
    line: 13,
    endColumn: 5,
    endLine: 13,
    ruleId: 'no-unused-vars',
    message: `'y' is assigned a value but never used.`,
    nodeType: '',
    severity: 0
  };
  const ES_LINT_MESSAGE_WARNING = {
    column: 10,
    line: 15,
    message: `'y' is assigned a value but never used.`,
    nodeType: '',
    severity: 1
  };
  const ES_LINT_RESULT_INDEX = {
    filePath: 'src/index.ts',
    messages: [ES_LINT_MESSAGE_ERROR],
    errorCount: 1,
    warningCount: 0,
    fixableErrorCount: 0,
    fixableWarningCount: 0
  };
  const ES_LINT_RESULT_ANOTHER = {
    filePath: 'src/another.ts',
    messages: [ES_LINT_MESSAGE_WARNING],
    errorCount: 0,
    warningCount: 1,
    fixableErrorCount: 0,
    fixableWarningCount: 0
  };
  const ES_LINT_RESULT_ADDITIONAL = {
    filePath: 'src/additional.ts',
    messages: [ES_LINT_MESSAGE_ERROR],
    errorCount: 1,
    warningCount: 0,
    fixableErrorCount: 0,
    fixableWarningCount: 0
  };
  const ES_LINT_REPORT_A = {
    results: [ES_LINT_RESULT_INDEX, ES_LINT_RESULT_ANOTHER],
    errorCount: 1,
    warningCount: 1,
    fixableErrorCount: 0,
    fixableWarningCount: 0
  };
  const ES_LINT_REPORT_B = {
    results: [ES_LINT_RESULT_ADDITIONAL],
    errorCount: 1,
    warningCount: 1,
    fixableErrorCount: 0,
    fixableWarningCount: 0
  };
  const ES_LINT_REPORTS = [ES_LINT_REPORT_A, ES_LINT_REPORT_B];

  it.each([[ES_LINT_REPORTS]])(
    'creates Issues from EsLint Reports: %p',
    reports => {
      const issues = createIssuesFromEsLintReports(reports);

      expect(issues).toMatchSnapshot();
    }
  );
});
