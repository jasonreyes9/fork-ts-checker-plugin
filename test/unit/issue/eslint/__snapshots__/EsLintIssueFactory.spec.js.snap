// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UNIT] issue/eslint/EsLintIssueFactory creates Issues from EsLint Reports: [[Object], [Object]] 1`] = `
Array [
  Object {
    "character": 0,
    "code": "no-unused-vars",
    "file": "src/additional.ts",
    "line": 13,
    "message": "'y' is assigned a value but never used.",
    "origin": "eslint",
    "severity": "error",
  },
  Object {
    "character": 10,
    "code": "[unknown]",
    "file": "src/another.ts",
    "line": 15,
    "message": "'y' is assigned a value but never used.",
    "origin": "eslint",
    "severity": "warning",
  },
  Object {
    "character": 0,
    "code": "no-unused-vars",
    "file": "src/index.ts",
    "line": 13,
    "message": "'y' is assigned a value but never used.",
    "origin": "eslint",
    "severity": "error",
  },
]
`;
